system_tests/pruning_test.go

```diff
@@ -105,15 +105,43 @@ func TestPruning(t *testing.T) {
 			Fatal(t, "The db doesn't have less entries after pruning then before. Before:", chainDbEntriesBeforePruning, "After:", chainDbEntriesAfterPruning)
 		}
 	}()
-	builder := NewNodeBuilder(ctx).DefaultConfig(t, true)
-	builder.dataDir = dataDir
-	cancel = builder.Build(t)
-	defer cancel()
 
-	builder.L2Info.GenerateAccount("User2")
+	testClient, cleanup := builder.Build2ndNode(t, &SecondNodeParams{stackConfig: builder.l2StackConfig})
+	defer cleanup()
+
+	currentBlock := uint64(0)
+	// wait for the chain to catch up
+	for currentBlock < lastBlock {
+		currentBlock, err = testClient.Client.BlockNumber(ctx)
```

Comment:
Should there be a small delay here to prevent spinning for loop?

Reply:
fixed, thanks!